lib/msf/core/feature_manager.rb

```diff
@@ -66,12 +72,19 @@ def set(name, value)
 
       @flag_lookup[name][:user_preference] = value
 
-      if name == WRAPPED_TABLES
+      case name
+      when WRAPPED_TABLES
         if value
           Rex::Text::Table.wrap_tables!
         else
           Rex::Text::Table.unwrap_tables!
         end
+      when FULLY_INTERACTIVE_SHELLS
+        if value
+          Rex::Post::Meterpreter::Ui::Console::CommandDispatcher::Stdapi::Sys.enable_fully_interactive!
+        else
+          Rex::Post::Meterpreter::Ui::Console::CommandDispatcher::Stdapi::Sys.disable_fully_interactive!
+        end
```

Comment:
Let's look at a way of doing this without the side effects, so we don't need to load all of the meterpreter classes when loading the feature manager :+1: